x-pack/plugins/enterprise_search/public/applications/shared/http/http_logic.ts

```diff
@@ -61,11 +63,9 @@ export const HttpLogic = kea<MakeLogicType<HttpValues, HttpActions>>({
         responseError: async (httpResponse) => {
           if (isEnterpriseSearchApi(httpResponse)) {
             const hasErrorConnecting = httpResponse.response!.headers.get(ERROR_CONNECTING_HEADER);
-
             if (hasErrorConnecting === 'true') {
-              actions.setErrorConnecting(true);
-            } else {
-              actions.setErrorConnecting(false);
```

Comment:
You might have noticed you can't _unset_ `errorConnectingMessage`.  Right now there's no actions the user can take on the error connecting screen, so there's no way for another Enterprise Search API request to take place.  We're not doing any sort of polling as far as I could tell that might trigger this.  Once the user has validated their connection to Enterprise Search is setup correctly, they'll refresh the page, which refreshes the app, which will set this back to its default on load if we can establish a connection.